src/Tests/ExecutionPathTracerTests.cs

```diff
@@ -558,6 +558,51 @@ public void BigTest()
         }
     }
 
+    [TestClass]
+    public class CanonTests : ExecutionPathTracerTests
+    {
+        [TestMethod]
+        public void ApplyToEachTest()
+        {
+            var path = GetExecutionPath("ApplyToEachCirc");
+            var qubits = new QubitDeclaration[]
+            {
+                new QubitDeclaration(0),
+                new QubitDeclaration(1),
+                new QubitDeclaration(2),
+            };
+            var operations = new Operation[]
+            {
+                new Operation()
+                {
+                    Gate = "H",
+                    Targets = new List<Register>() { new QubitRegister(0) },
+                },
+                new Operation()
+                {
+                    Gate = "H",
+                    Targets = new List<Register>() { new QubitRegister(1) },
+                },
+                new Operation()
+                {
+                    Gate = "H",
+                    Targets = new List<Register>() { new QubitRegister(2) },
+                },
+                new Operation()
+                {
+                    Gate = "ResetAll",
```

Comment:
I wonder if `ResetAll` should be marked as composite too?

Reply:
We can do that too. I can create a PR in `qsharp-runtime` to address this and make the necessary changes in a separate PR.

Reply:
Made a PR for that: https://github.com/microsoft/qsharp-runtime/pull/318